### PR TITLE
fix(api): strip api/0/ prefix and exclude NodeSystemError integration (CLI-K1)

### PR DESCRIPTION
**Pull request by BYK**
Fixes two bugs surfaced by [CLI-K1](https://sentry.sentry.io/issues/7355494310/):

**1. Double `/api/0/` prefix in `sentry api` command**

When users include the `/api/0/` prefix in the endpoint argument (e.g. `sentry api /api/0/projects/my-org/my-project/environments/`), the URL construction doubled it to `/api/0/api/0/...`. `normalizeEndpoint()` now strips the prefix automatically.

**2. `util.getSystemErrorMap()` crash on Bun**

The Sentry SDK's `NodeSystemError` integration calls `util.getSystemErrorMap()` which Bun does not implement, causing the SDK's event processing pipeline to crash with a TypeError. This integration is now excluded from the default integrations set — it only adds extra context for Node.js system errors and is not needed for CLI telemetry.

### CODE REVIEW

File: src/commands/api.ts [hunk 1]
```diff
@@ -87,7 +87,16 @@ export function normalizeEndpoint(endpoint: string): string {
...
-  const trimmed = endpoint.startsWith("/") ? endpoint.slice(1) : endpoint;
+  let trimmed = endpoint.startsWith("/") ? endpoint.slice(1) : endpoint;
+
+  // Strip api/0/ prefix if user accidentally included it — the base URL
+  // already includes /api/0/, so keeping it would produce a doubled path
+  // like /api/0/api/0/... (see CLI-K1).
+  // Also strip bare "api/0" to maintain idempotency: without this,
+  // "api/0" → "api/0/" → "" which breaks f(f(x)) === f(x).
+  if (trimmed.startsWith("api/0/") || trimmed === "api/0") {
+    trimmed = trimmed.slice(trimmed.startsWith("api/0/") ? 6 : 5);
```
Comment (BYK) [inline src/commands/api.ts:97]: We should probably issue a warning about this?

File: src/lib/telemetry.ts [hunk 2]
```diff
@@ -260,6 +260,7 @@ const EXCLUDED_INTEGRATIONS = new Set([
   "ContextLines", // Reads source files - we rely on uploaded sourcemaps instead
   "LocalVariables", // Captures local variables - adds significant overhead
   "Modules", // Lists all loaded modules - unnecessary for CLI telemetry
+  "NodeSystemError", // Uses util.getSystemErrorMap() which Bun does not implement (CLI-K1)
```
Comment (BYK) [inline src/lib/telemetry.ts]: We should patch the SDK for this so we keep getting proper error reports.

File: src/commands/api.ts (hunk 1 above)
Comment (BYK) [reply src/commands/api.ts:97]: Done — added `log.warn()` when stripping the prefix so users see what happened.

File: src/lib/telemetry.ts (hunk 2 above)
Comment (BYK) [reply src/lib/telemetry.ts]: Done — instead of excluding the integration, I now wrap its `processEvent` in a try-catch. On runtimes that support `getSystemErrorMap` (Node.js or future Bun versions) the system error context is still attached. On Bun, the error is caught and the event passes through unmodified.

File: src/commands/api.ts [hunk 3]
```diff
@@ -87,7 +87,16 @@ export function normalizeEndpoint(endpoint: string): string {
...
+  }
```
[the remark this answers is not in the document]
Comment (BYK) [reply src/commands/api.ts:98]: Good catch — added `trimmed.startsWith("api/0?")` to the condition. Also added unit tests for this edge case.

File: src/lib/telemetry.ts [hunk 4]
```diff
@@ -260,9 +262,41 @@ const EXCLUDED_INTEGRATIONS = new Set([
...
+/**
+ * Patch the NodeSystemError integration's processEvent to catch missing
+ * `util.getSystemErrorMap()` on Bun. Without this, the SDK crashes during
+ * event processing instead of sending the error report (CLI-K1).
+ *
+ * We wrap rather than exclude the integration so that on runtimes (or future
+ * Bun versions) that implement `getSystemErrorMap`, system error context is
+ * still attached.
+ */
+function patchNodeSystemErrorIntegration(
```
Comment (BYK) [inline src/lib/telemetry.ts]: I'd say we should either remove this integration as we lived without it so far without any issues or check for the existence of the method at patch time and avoid try-catch and just make it a noop when the `getSystemErrorMap` is not available.

File: src/commands/api.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (BYK) [reply src/commands/api.ts:98]: And what would be the `/api/0?` URL would be useful for? That's clearly an error so I don't think it needs to be accounted for?

[the remark this answers is not in the document]
Comment (BYK) [reply src/commands/api.ts:98]: Fair point — removed the `api/0?` handling. The user would get a clear API error from the doubled path, which is a better signal than silently fixing nonsensical input.

File: src/lib/telemetry.ts (hunk 4 above)
Comment (BYK) [reply src/lib/telemetry.ts]: Done — replaced the try-catch wrapper with an upfront check: `hasGetSystemErrorMap` is evaluated once at module load via `require("node:util")`. When `getSystemErrorMap` is missing (Bun), the `NodeSystemError` integration is filtered out entirely alongside the other excluded integrations. On Node.js it stays active.

File: src/lib/telemetry.ts [hunk 5]
```diff
@@ -262,6 +264,39 @@ const EXCLUDED_INTEGRATIONS = new Set([
...
+  }
+
+  integration.processEvent = (...args) => {
+    try {
+      return original(...args);
+    } catch {
+      // getSystemErrorMap is not available — skip enrichment, keep the event
+      return args[0];
+    }
+  };
```
[the remark this answers is not in the document]
Comment (BYK) [reply src/lib/telemetry.ts]: Moot — the try-catch wrapper has been replaced with an upfront `typeof util.getSystemErrorMap === "function"` check at module load. When it's missing, the integration is excluded entirely instead of being patched.

File: test/commands/api.property.test.ts [hunk 6]
```diff
@@ -211,6 +211,31 @@ describe("normalizeEndpoint properties", () => {
...
+
+  test("result never starts with api/0/ (CLI-K1)", async () => {
+    await fcAssert(
+      property(endpointArb, (endpoint) => {
+        const result = normalizeEndpoint(endpoint);
+        expect(result.startsWith("api/0/")).toBe(false);
+      }),
+      { numRuns: DEFAULT_NUM_RUNS }
+    );
+  });
```
[the remark this answers is not in the document]
Comment (BYK) [reply test/commands/api.property.test.ts:238]: Correct that `endpointArb` can't generate `"0"` segments. However, the companion test `"api/0/ prefix is always stripped"` explicitly prepends `api/0/` to generated paths, which does exercise the stripping code. The `"result never starts with api/0/"` test serves as a defense-in-depth invariant — it's vacuously true for the current generator but would catch regressions if someone introduced a code path that accidentally added the prefix.

File: src/commands/api.ts [hunk 7]
```diff
@@ -1160,6 +1168,14 @@ export const apiCommand = buildCommand({
     const { stdin } = this;
 
     const normalizedEndpoint = normalizeEndpoint(endpoint);
+
+    // Warn if the user included the /api/0/ prefix (CLI-K1)
+    const bare = endpoint.startsWith("/") ? endpoint.slice(1) : endpoint;
+    if (bare.startsWith("api/0/") || bare === "api/0") {
```
[the remark this answers is not in the document]
Comment (BYK) [reply src/commands/api.ts]: Good point — replaced the duplicated condition with a length comparison. Since `normalizeEndpoint` only adds at most 1 char (trailing slash), a shorter output indicates the prefix was stripped.
